libs/header/sync/ranges.go

```diff
@@ -33,6 +33,21 @@ func (rs *ranges[H]) head() H {
 	return head.Head()
 }
 
+// cached returns all headers sub-ranges of headers that could be found in between the requested range.
```

Comment:
```suggestion
// cached returns all headers contiguous sub-ranges of headers that could be found within the requested range.
```
is this what you meant?

Reply:
yes. but I removed it